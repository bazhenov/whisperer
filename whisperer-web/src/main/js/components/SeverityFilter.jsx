import React, {Component} from "react";

const SeverityVariant = ({title, isSelected, clickHandler, disabled }) => {
	return isSelected ? <span className="choosed-severity label label-primary">{title}</span> :
		disabled ? <span className="severity-button">{title}</span> :
			<a className="severity-button" onClick={clickHandler} href="#">{title}</a>
};

export default class SeverityFilter extends Component {

	constructor(props) {
		super(props);
		this.handleClick = this.handleClick.bind(this);
	}

	handleClick(e) {
		e.preventDefault();
		this.props.updateSeverity(
			e.target.text
		)
	}

	render() {
		const { currentSeverity, disabled } = this.props;
		const buttons = ['INFO', 'WARN', 'ERROR', 'DEBUG', 'TRACE'].map(severity => {
			return <SeverityVariant key={severity}
															clickHandler={this.handleClick}
															title={severity}
															isSelected={currentSeverity === severity}
															disabled={disabled} />
		});
		return <div className="form-group">
			<div><label>Severity</label></div>
			{ buttons }
		</div>
	}
}