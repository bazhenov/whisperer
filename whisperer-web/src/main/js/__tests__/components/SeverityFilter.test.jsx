import React from "react";
import SeverityFilter from "../../components/SeverityFilter.jsx";
import renderer from "react-test-renderer";
import {mount} from "enzyme";

describe('SeverityFilter', () => {
	const props = {
		currentSeverity: "INFO",
		updateSeverity: () => {},
		disabled: false
	};

	it('rendering', () => {
		let tree = renderer.create(
			<SeverityFilter { ...props } />
		).toJSON();
		expect(tree).toMatchSnapshot();

		tree = renderer.create(
			<SeverityFilter { ...props } disabled={true} />
		).toJSON();
		expect(tree).toMatchSnapshot()
	});

	it('click handler', done => {
		const chooseSeverity = (severity) => {
			expect(severity).toBe('WARN');
			done();
		};
		const enzymeWrapper = mount(
			<SeverityFilter { ...props } updateSeverity={chooseSeverity} />
		);
		enzymeWrapper.find('.severity-button').first().simulate('click');
	});
});