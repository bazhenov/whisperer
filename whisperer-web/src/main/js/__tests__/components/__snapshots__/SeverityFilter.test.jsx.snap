// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeverityFilter rendering 1`] = `
<div
  className="form-group"
>
  <div>
    <label>
      Severity
    </label>
  </div>
  <span
    className="choosed-severity label label-primary"
  >
    INFO
  </span>
  <a
    className="severity-button"
    href="#"
    onClick={[Function]}
  >
    WARN
  </a>
  <a
    className="severity-button"
    href="#"
    onClick={[Function]}
  >
    ERROR
  </a>
  <a
    className="severity-button"
    href="#"
    onClick={[Function]}
  >
    DEBUG
  </a>
  <a
    className="severity-button"
    href="#"
    onClick={[Function]}
  >
    TRACE
  </a>
</div>
`;

exports[`SeverityFilter rendering 2`] = `
<div
  className="form-group"
>
  <div>
    <label>
      Severity
    </label>
  </div>
  <span
    className="choosed-severity label label-primary"
  >
    INFO
  </span>
  <span
    className="severity-button"
  >
    WARN
  </span>
  <span
    className="severity-button"
  >
    ERROR
  </span>
  <span
    className="severity-button"
  >
    DEBUG
  </span>
  <span
    className="severity-button"
  >
    TRACE
  </span>
</div>
`;
